connectors/jdk-connector/src/main/java/org/glassfish/jersey/jdk/connector/internal/SslFilter.java

```diff
@@ -416,12 +419,8 @@ private boolean doHandshakeStep(ByteBuffer networkData) {
                         }
 
                         case FINISHED: {
-                            /* According to SSLEngine javadoc FINISHED status can be returned only in SSLEngineResult,
-                            but just to make sure we don't end up in an infinite loop when presented with an SSLEngine
-                            implementation that does not respect this:*/
-                            stepFinished = true;
-                            handshakeFinished = true;
-                            break;
+                            throw new IllegalStateException("Trying to handshake, but SSL engine not in HANDSHAKING state."
```

Comment:
I would advice to move direct text into localization.properties and use LocalizableMessageFactory instead.

Reply:
@senivam although that is unreachable (in theory), I decided to keep it as it was. It will not throw exception.

I added the localization property for the case of NOT_HANDSHAKING, because it was throwing the same exception.

I also modified the test to force to test all supported SSL protocols. For the case of JDK11 the TLSv13 is included.

Reply:
I just verified with Wireshark that we need to execute the tests in different JVMs. So I had to set `<reuseForks>false</reuseForks>` and move each protocol to a different test class. It seems when you use `@Parametrized` tests are executed in the same JVM.